core/pkg/monitor/monitor.go

```diff
@@ -191,16 +194,21 @@ func (sm *SystemMonitor) Do() {
 		go sm.Monitor(asset)
 	}
 
-	// probe the asset information
-	go func() {
-		systemInfo := sm.Probe()
-		if systemInfo != nil {
-			sm.extraWork.AddRecordOrCancel(
-				sm.ctx.Done(),
-				makeMetadataRecord(systemInfo),
-			)
-		}
-	}()
+	// probe the asset information.
+	// NOTE: in notebook environments, we stop the system monitor when a cell has finished
+	// executing, and start it again when a new cell is executed. There is no need to probe
+	// the system information every time the system monitor is started.
```

Comment:
So we are calling `SystemMonitor.Do()` multiple times? Is that really what we want? I'd expect this to have a `Start` and `Finish` method like on other system components.

It's not clear to me why we only wrap this part in `probeOnce` and not the rest of the code.

Reply:
we should not be collecting system metrics when no cell is being executed in notebook environments, this is what Request_Pause and Request_Resume are for (also, we pause/resume the run timer). we do want to resume monitoring hw resources, but we shouldn't be probing them every time.